controllers/imagepolicy_controller.go

```diff
@@ -89,22 +91,26 @@ func (r *ImagePolicyReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 
 	policy := pol.Spec.Policy
 
+	var latest string
+	var err error
 	switch {
 	case policy.SemVer != nil:
-		latest, err := r.calculateLatestImageSemver(&policy, repo.Status.CanonicalImageName)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		if latest != "" {
-			pol.Status.LatestImage = repo.Spec.Image + ":" + latest
-			err = r.Status().Update(ctx, &pol)
-		}
-		return ctrl.Result{}, err
+		latest, err = r.calculateLatestImageSemver(&policy, repo.Status.CanonicalImageName)
```

Comment:
I wonder if we couldn't break these out into implementations of a _Policy_ interface, with a simple map translation here, which would make it easier to test the policies "standalone" ?

Reply:
What does "map translation" mean here -- dispatch via a map of policy name -> implementation?

Reply:
I think that makes sense. I'm going to try to come up with something.

Reply:
@squaremo yes

Reply:
I've moved the logic to an internal `policy` package and defined there the policy types and a factory to resolve the type based on the `ImagePolicyChoice` object. I think this is a lot better than the switch and allows us to extend further if needed without involving any controller logic changes.